transform/transform.go

```diff
@@ -78,12 +84,53 @@ func (ti *transformInstruction) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
-// transform runs the instructions in this object returning the new transformed value or an error if unable to.
-// It handles the logic for finding the value to be transformed and chaining the Operations.
-// It will not error if the value is not found, rather it returns nil for the value.
-// If a conversion or operation fails an error is returned.
-func (ti *transformInstruction) transform(in interface{}, fieldType string, modifier pathModifier) (interface{}, error) {
-	path := ti.jsonPath
+func (ti *transformInstruction) xmlTransform(in interface{}, fieldType string, modifier pathModifier) (interface{}, error) {
+	path := ti.XMLPath
+	if modifier != nil {
+		path = modifier(path)
+	}
+
+	node, ok := in.(*xmlquery.Node)
+	if !ok {
+		return nil, errors.New("Error converting input to *xmlquery.Node")
+	}
+
+	xmlNode := xmlquery.Find(node, path)
+	if xmlNode == nil {
+		return nil, nil
+	}
+	rawValue := xmlNode
```

Comment:
actually. Followup question is if that can ever return nil - and I actually see no.

Reply:
do we want to add a nil check here? I see one in the other method to save some time on trying to convert nil